### DISCUSSION TITLE
Communication between Frontend App, and Node Backend

### DISCUSSION THREAD

**Discussion by lcainswebdeveloper** [nextauthjs/next-auth#4358] [Help]
### Question 💬

Hi! Apologies if this is a duplicate, but I haven't been able to find the answer I'm looking for to what is hopefully a straight forward issue to solve.

I'm building an app which has a node js backend, a React/Next front end and a shared mongo database.
Ultimately what I want to do is use NextAuth with the FE code to authenticate a user (currently either by Github or Google), create a user in a users table (which it does), and then on subsequent requests to my back end, pass a means of securely identifying the authenticated user (so that I can log the user id against created articles, categories etc).

The code I have is as below so can someone please help my identify what the best practise would be please for what i'm trying to do? I was hoping to use session.access_token but it continually returns undefined.

Thanks very much :-)

```js
/** @format */

import NextAuth from "next-auth";
import GoogleProvider from "next-auth/providers/google";
import GithubProvider from "next-auth/providers/github";
import { MongoDBAdapter } from "@next-auth/mongodb-adapter";
import clientPromise from "./../../../lib/mongodb";
// For more information on each option (and a full list of options) go to
// https://next-auth.js.org/configuration/options
export default NextAuth({
    adapter: MongoDBAdapter(clientPromise),
    session: { strategy: "jwt" },
    // https://next-auth.js.org/configuration/providers/oauth
    providers: [
        GithAubProvider({
            clientId: process.env.GITHUB_ID,
            clientSecret: process.env.GITHUB_SECRET,
        }),
        GoogleProvider({
            clientId: process.env.GOOGLE_ID,
            clientSecret: process.env.GOOGLE_SECRET,
        }),
    ],
    jwt: {
        encryption: true,
    },
    theme: {
        colorScheme: "light",
    },
    callbacks: {
        async jwt(data) {
            const { token, account } = data;
            token.userRole = "admin";
            if (account?.access_token) {
                token.access_token = account.access_token;
            }
            return token;
        },
        async session(data) {
            const { session, token, user } = data;
            if (token?.access_token) {
			session.access_token = token.access_token;
	}
            if (user && user.id) {
                session.user_id = user.id;
            }
            return session;
        },
    },
});
```

### How to reproduce ☕️

Please see code snippet above

### Contributing 🙌🏽

Yes, I am willing to help answer this question in a PR

**Comment by lcainswebdeveloper** [comment]
Hi, sorry to bump but is anyone able to assist here please?

**Reply by ThangHuuVu** [reply]
Could you see `token.access_token` in the JWT callback? 
Also, which version of NextAuth.js are you on?
